apps/astarte_housekeeping/lib/astarte_housekeeping/c_system.ex

```diff
@@ -46,61 +51,54 @@ defmodule CSystem do
           {:error, :timeout}
         else
           Process.sleep(millis)
-          wait_schema_agreement(conn, timeout - millis)
+          wait_schema_agreement(timeout - millis)
         end
-
-      any_other ->
-        any_other
     end
   end
 
-  def schema_versions(conn) do
-    with {:ok, local_version} <- query_local_schema_version(conn),
-         {:ok, peers_versions} <- query_peers_schema_versions(conn) do
+  def schema_versions() do
+    with local_version <- query_local_schema_version(),
+         peers_versions <- query_peers_schema_versions() do
       {:ok, Enum.uniq([local_version | peers_versions])}
     end
```

Comment:
this with is just assigning variables now

```suggestion
    local_version = query_local_schema_version()
    peers_versions = query_peers_schema_versions()
    Enum.uniq([local_version | peers_versions])
```